File: E_Red_Scarf/v1.js
const fs = require('fs')
const inputs = fs.readFileSync('/dev/stdin', 'utf8').trim().split(/\s+/).map(s => parseInt(s))

const n = inputs.shift()
const arr = inputs.slice(0, n)
const allXor = arr.reduce((a, b) => a ^ b, 0)
const results = arr.map(a => allXor ^ a)

console.log(results.join(' '))

File: beginner_contest_171/E_Red_Scarf/v1.cpp
#include<iostream>
#include<vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    unsigned int allXor = 0;
    vector<unsigned int> arr(n);
    for (int i = 0; i < n; i++) {
        unsigned int a;
        cin >> a;
        allXor = allXor ^ a;
        arr[i] = a;
    }

    for (int i = 0; i < n - 1; i++) {
        cout << (arr[i] ^ allXor) << " ";
    }

    cout << (arr[n - 1] ^ allXor);
}
